src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XNodeReader.cs

```
@@ -552,25 +552,26 @@ public override string GetAttribute(string localName, string namespaceName)
             return null;
         }
 
+        // TODO-NULLABLE: decide if base signature should be switched to return string?
```

do we throw elsewhere or also return null?

---

> or also return null

Not in plain sight as in here. The other two overloads does return `string?`.
https://github.com/dotnet/runtime/blob/f0461cb51032e45cbcabf2152e0c82c8c5a890bc/src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs#L619-L626

---

those are different, `GetAttribute(int)` takes index and should throw when you request invalid index, for other two you check by name and it is more frequent scenario to check if attribute is present

---

Does this still need to be resolved?

---

I think this is ok as is